### PR TITLE
Fix gaussian vhmm

### PR DESCRIPTION
**Pull request by blckmaxima**
Addressing subtle issues with the Variational Gaussian HMM

### CODE REVIEW

**Comment by blckmaxima** [comment]
I am expecting this to fail, and am tracking down a final bug...

**Comment by blckmaxima** [comment]
And yes, I'll be deleting some `print()` before merging :-p

**Comment by blckmaxima** [comment]
@anntzer What are your thoughts on the change I made in `base.py` about "strict".    In testing this week, the best I can understand is the lower bound does not increase due  to floating point rounding error. 

Another option is to actually back out the notion of "strict", and let end-users validate the model training history themselves.

**Comment by anntzer** [comment]
We used not to check at all (AFAICT) and you introduced the check in the vhmm PR.  I guess the check is useful, but allowing floating point slop is reasonable (sqrt(eps) seems like a reasonable allowance on general grounds), and choosing e.g. to warn instead of raising would probably be good too.  I would then get rid of the "strict" API and let users rely on the stdlib warnings API to control the warnings.

**Comment by blckmaxima** [comment]
Good call on backing out the `strict` parameter I added.

**Comment by blckmaxima** [comment]
@anntzer This should be good to go.

File: lib/hmmlearn/vhmm.py [hunk 1]
```diff
@@ -784,7 +787,7 @@ def _compute_lower_bound(self, log_prob):
         if self.covariance_type != "full":
             scale_posterior_ = fill_covars(self.scale_posterior_,
                     self.covariance_type, self.n_components, self.n_features)
-            scale_prior_ = fill_covars(self.scale_posterior_,
+            scale_prior_ = fill_covars(self.scale_prior_,
```
Comment (anntzer) [inline lib/hmmlearn/vhmm.py:790]: Is this effectively the only material change?  (I trust you on the formulas...)

Comment (blckmaxima) [reply lib/hmmlearn/vhmm.py:790]: I have a follow-on MR that will then reuse most of this in a GMM, and I've been comparing the formulas across a few different papers.

**Pull request merged**